paddle/fluid/framework/op_registry.h

```diff
@@ -189,6 +206,15 @@ class OpKernelRegistrar : public Registrar {
       __attribute__((unused)) =                                   \
           TouchOpKernelRegistrar_##op_type##_##LIBRARY_TYPE()
 
+#define USE_OP_DEVICE_KERNEL_EXTEND(op_type, LIBRARY_TYPE, LAYOUT)           \
```

Comment:
Why do you `USE_OP_DEVICE_KERNEL_EXTEND `? I could not find the usage of it.

Reply:
Please have a look at this [code](https://github.com/PaddlePaddle/Paddle/pull/11098/files#diff-2f55eef7dba89ace8cdce5f5726dbee6R160)

Reply:
Yes, that is a specific way for MKLDNN. This is the way to tell the MKLDNN's from the CPU's platform. These functions that we see are a little bit different, therefore I can't use any exist functions. Summary, this way to discern these operators gives us an ability to register the new MKLDNN's operator which has a support of layout.

Reply:
After discuss with @tensor-tang , we think that currently all mkldnn kernels have only one layout `MKLDNN`, so we can use `LIBRARY` to mark and distinguish them, mkldnn_kernels can be choosed by this library flag, data transform can use the `tensor.layout_` to decide if it need to do transform. 

It seems that currently all our work can be done without register layout to op_kernel_registry,  so can we just use LIBRARY for now and move forward on our work?